components/fingerprint_grow.rst

```diff
@@ -0,0 +1,530 @@
+Grow Fingerprint Reader
+=======================
+
+.. seo::
+    :description: Instructions for setting up Grow Fingerprint Reader integration in ESPHome.
+    :image: fingerprint.png
+
+The ``fingerprint_grow`` component allows you to use your R307, R503, ZFM-20, ... fingerprint sensors with ESPHome.
+
+.. figure:: images/r307-full.jpg
+    :align: center
+    :width: 50.0%
+
+    R307/ZFM-20 Optical Fingerprint Reader (`datasheet <https://cdn-shop.adafruit.com/datasheets/ZFM+user+manualV15.pdf>`__, `Adafruit <https://www.adafruit.com/product/751>`__). Image by `Adafruit <https://www.adafruit.com/product/751>`__.
+
+.. figure:: images/r503-full.jpg
+    :align: center
+    :width: 50.0%
+
+    R503 Capacitive Fingerprint Reader with Bi-Color Aura LED (`datasheet <https://cdn-shop.adafruit.com/product-files/4651/4651_R503+fingerprint+module+user+manual.pdf>`__, `Adafruit <https://www.adafruit.com/product/4651>`__). Image by `Adafruit <https://www.adafruit.com/product/4651>`__.
+
+Overview
+--------
+
+The reader can be powered by the 3.3V output of an NodeMCU. As the communication with the reader is done using UART (default baud rate is 57600), you need to have an :ref:`UART bus <uart>` in your configuration with the ``rx_pin`` connected to the reader's ``TX`` and the ``tx_pin`` connected to the reader's ``RX``.
+
+If available on your reader model, it's recommended to connect 3.3VT (touch induction power supply) to 3.3V; WAKEUP (finger detection signal) to a free GPIO pin and define it with the ``sensing_pin`` option to allow the polling function to quickly return when there's no finger on the reader.
```

Comment:
@loongyh thinking about this again, maybe a better way is to connect the touch induction power supply to a free GPIO pin in pull-up mode and WAKEUP to GND? Not sure, and my reader is already setup in a way that's hard for me to test.

Reply:
You mean GPIO pin in INPUT_PULLUP mode? Hmm how is that better? Will the input pullup be able to source enough current to supply power to the touch induction?

Reply:
My guess is that this could lead to lower consumption. I didn't measure the resistance of that circuit, but with an `INPUT_PULLUP` setup the current would be minimal.
Once again, this isn't based on measurements but on guesses :)

Reply:
Noted, will test it out and update soon.